.github/workflows/pr.yml

```diff
@@ -23,23 +23,36 @@ jobs:
         git status
         diff_count=$(git status --porcelain=v1 2>/dev/null | wc -l)
         exit $diff_count
+    ##
+    ## The following process first compiles and packages (dist) the tla2tools.jar file. Next, it 
+    ## runs the test suite in parallel, aborting the run and failing the workflow upon the first
+    ## test failure (test.halt=true). Regardless of any test failures, a JUnitReport is generated.
+    ## Both the report and the test results (in XML format) are uploaded as workflow artifacts for
+    ## potential local download and further analysis. Additionally, mikepenz/action-junit-report
+    ## publishes the stack traces from the test results. If desired, the test.halt setting can be
+    ## removed to allow the test suite to run to completion.
+    ## (The same sequence of steps is executed by main.yml)
+    ## 
     - name: Build tools and unit tests
-      run: ant -f tlatools/org.lamport.tlatools/customBuild.xml compile compile-test dist
+      run: ant -f tlatools/org.lamport.tlatools/customBuild.xml info compile compile-test dist
     - name: Run unit tests
-      run: |
-        set +e # Ensure UT failure does not immediately terminate CI
-        set -o pipefail # Ensure UT failure is propagated through tee command
-        ant -f tlatools/org.lamport.tlatools/customBuild.xml test 2>&1 | tee test-output.txt
-        ut_exit_code=$?
-        echo "Unit test exit code: $ut_exit_code"
-        echo "UT_EXIT_CODE=$ut_exit_code" >> $GITHUB_ENV # Export UT result to CI env var
-    - name: Summarize unit tests
-      run: |
-        echo "Unit test exit code: $UT_EXIT_CODE"
-        python .github/scripts/parse-unit-test-output.py        \
-          test-output.txt                                       \
-          tlatools/org.lamport.tlatools/target/surefire-reports
-        exit $UT_EXIT_CODE
+      run: ant -f tlatools/org.lamport.tlatools/customBuild.xml -Dtest.halt=true test
```

Comment:
Is `-Dtest.halt=true` desirable? We might want to see all failing tests instead of just the first one. Because JUnit runs tests in an arbitrary order, the first failing test is not even deterministic.

Reply:
This applies to the similar line in `main.yml` as well.

Reply:
I'm not married to it, but I've set `test.halt=true` in `pr.yml` and changed it to `test.halt=false` in `main.yml`. However, one can override `test.halt` per run via a workflow input in `main.yml` (I don't believe inputs work for PR workflows).

https://github.com/tlaplus/tlaplus/pull/954/commits/acbf907949a4d98aa566f1411f5262b96e3a6735